packages/docusaurus/src/server/configValidation.ts

```diff
@@ -81,8 +80,33 @@ export const ConfigSchema = Joi.object({
     }),
   ),
   tagline: Joi.string(),
-}).messages({
-  'object.unknown': `The field {{#label}} is not recognized in ${CONFIG_FILE_NAME}.
-If you still want these fields to be in your configuration, put them in the 'customFields' attribute.
-See https://v2.docusaurus.io/docs/docusaurus.config.js/#customfields`,
 });
+
+export function validateConfig(
+  config: Partial<DocusaurusConfig>,
+): DocusaurusConfig {
+  const {error, value} = ConfigSchema.validate(config, {
+    abortEarly: false,
+  });
+  if (error) {
+    const unknownFields = error.details.reduce((formatedError, err) => {
+      if (err.type === 'object.unknown') {
+        return `${formatedError}"${err.path}",`;
+      }
+      return formatedError;
+    }, '');
+    let formatedError = error.details.reduce(
+      (accumalatedErr, err) =>
+        err.type !== 'object.unknown'
+          ? `${accumalatedErr}${err.message}\n`
+          : accumalatedErr,
+      '',
+    );
+    formatedError = unknownFields
+      ? `These field(s) [${unknownFields}] are not recognized in ${CONFIG_FILE_NAME}.\nIf you still want these fields to be in your configuration, put them in the 'customFields' attribute.\nSee https://v2.docusaurus.io/docs/docusaurus.config.js/#customfields${formatedError}`
+      : formatedError;
+    throw new Error(formatedError);
+  } else {
+    return value;
+  }
+}
```

Comment:
oh :/ that's complicated. Surprised that Joi does not permit more easily to extract validation error datas.

I'd rather revert to abortEarly: true for now.

What do you think?

Reply:
I think same

Reply:
should i revert?

Reply:
Let's keep this for now, it seems to work, we could simplify this later if this becomes a problem or there are unsupported edge cases